pulpcore/cli/common/generic.py

```diff
@@ -414,14 +414,9 @@ def _href_callback(
 def _version_callback(
     ctx: click.Context, param: click.Parameter, value: t.Optional[int]
 ) -> t.Optional[int]:
-    entity_ctx = ctx.find_object(PulpEntityContext)
-    assert entity_ctx is not None
-    repository_ctx = ctx.find_object(PulpRepositoryContext)
-    assert repository_ctx is not None
-    if value is not None:
-        entity_ctx.pulp_href = f"{repository_ctx.entity['versions_href']}{value}/"
-    else:
-        entity_ctx.pulp_href = repository_ctx.entity["latest_version_href"]
+    repository_version_ctx = ctx.find_object(PulpRepositoryVersionContext)
+    assert repository_version_ctx is not None
+    repository_version_ctx.entity = {"number": value}
```

Comment:
That is it? So the solution was there ready to be used all along...
Please add that test i gave you?

Reply:
Interestingly this doesn't break the scenario where you don't supply the `--version` option because versions are always listed highest first. It does add the message `Not all X entries were shown` to the output, but the behavior is still correct.

Reply:
What are you telling me? We do the lookup with limit=1 and so we incidentally only get the latest by not using the number filter?

Reply:
Correct. The behavior before was that if you didn't supply the `--version` option the version callback would set the `RepositoryVersionContext` to the repository's latest version href. Now if you don't supply the option the `find` method will just use the first version returned because of `limit=1` and it just so happens that we always return versions newest/latest first.

Reply:
@mdellweg I saw your uniqueness check PR, does implicit latest version in my code ({"number": None}) trigger the exception you added for multiple found results?

Reply:
@mdellweg by entity getter you're not referring to PulpRepositoryVersionContext's get_version_context() function right? Since that is called only without arguments in this case, making us unable to use either -1 or None to get the latest version.